### PR TITLE
Fixes SI-5822 & SI-6305 - OSGi tests + fixes

### PR DESCRIPTION
**Pull request by jsuereth**
This pull requests brings:
- OSGi manfiest generation (via BND) into the ant build.  
- Felix/equinox tests against the defined bundles.  The "basic" test is enough to ensure that our packages resolve correctly.  
- A very simplistic reflection test.   Tests basic mirror usage.  Any more and we start breaking (which I'll be reporting as bugs as soon as the ability to write tests for them is in the 2.10.x branch).
- Refactoring scala.tools.nsc.io split package so that it's OSGi friendly.

Note: These changes should _not_ affect normal dev.  Only running ant test or dist.done will cause the extra bundle creation.   I'll have to look into ways to speed it up more in the future.

Review by @xeno-by and @paulp

Special thanks to @jamie-allen and @martiell for help on this.

### CODE REVIEW

File: src/reflect/scala/reflect/runtime/SynchronizedSymbols.scala [hunk 1]
```diff
@@ -123,7 +121,7 @@ trait SynchronizedSymbols extends internal.Symbols { self: SymbolTable =>
 
   trait SynchronizedClassSymbol extends ClassSymbol with SynchronizedTypeSymbol {
     override def associatedFile = synchronized { super.associatedFile }
-    override def associatedFile_=(f: AbstractFile) = synchronized { super.associatedFile_=(f) }
+    override def associatedFile_=(f: io.AbstractFile) = synchronized { super.associatedFile_=(f) }
```
Comment (xeno-by) [inline src/reflect/scala/reflect/runtime/SynchronizedSymbols.scala]: relative imports? ;)

File: src/reflect/scala/reflect/io/AbstractFile.scala [hunk 2]
```diff
@@ -13,6 +13,10 @@ import java.net.URL
 import scala.collection.mutable.ArrayBuffer
 
 /**
+ * An abstraction over files for use in the reflection/compiler libraries.
+ * 
+ * ''Note:  This is library is considered experimental and should not be used unless you know what you are doing.''
```
Comment (larsrh) [inline src/reflect/scala/reflect/io/AbstractFile.scala]: THIS. IS. LIBRARY.

Comment (jsuereth) [inline src/reflect/scala/reflect/io/AbstractFile.scala]: embarassing, will fix.

File: src/reflect/scala/reflect/runtime/SynchronizedSymbols.scala (hunk 1 above)
Comment (jsuereth) [inline src/reflect/scala/reflect/runtime/SynchronizedSymbols.scala]: can do.

**Comment by jsuereth** [comment]
PLS REBUILD ALL

**Comment by xeno-by** [comment]
lgtm

**Pull request merged**